casper/src/main/scala/coop/rchain/casper/BlockRandomSeed.scala

```diff
@@ -34,4 +56,114 @@ object BlockRandomSeed {
 
   def generateRandomNumber(blockRandomSeed: BlockRandomSeed): Blake2b512Random =
     Blake2b512Random(encode(blockRandomSeed))
+
+  def generateSplitRandomNumber(blockRandomSeed: BlockRandomSeed, index: Byte): Blake2b512Random =
+    generateRandomNumber(blockRandomSeed).splitByte(index)
+
+  def generateSplitRandomNumber(
+      blockRandomSeed: BlockRandomSeed,
+      index: Byte,
+      index2: Byte
+  ): Blake2b512Random =
+    generateRandomNumber(blockRandomSeed).splitByte(index).splitByte(index2)
+
+  def fromBlock(block: BlockMessage): Blake2b512Random = {
+    val seed = BlockRandomSeed(
+      block.shardId,
+      block.blockNumber,
+      PublicKey(block.sender),
+      block.preStateHash.toBlake2b256Hash
+    )
+    generateRandomNumber(seed)
+  }
+
+  def fromGenesis(block: BlockMessage): Blake2b512Random =
+    fromGenesis(block.shardId)
+
+  def fromGenesis(shardId: String): Blake2b512Random = {
+    val seed = BlockRandomSeed(
+      shardId,
+      GenesisRandomSeedBlockNumber,
+      GenesisRandomSeedPubKey,
+      EmptyBytesBlakeHash
+    )
+    generateRandomNumber(seed)
+  }
+
+  def nonNegativeMergeableTagName(
+      shardId: String
+  ): Par = {
+    // NonNegative contract is the 4th contract deployed in the genesis, start from 0. Index should be 3
+    val nonNegativeContractIndex: Byte = 3
+    val seed = BlockRandomSeed(
+      shardId,
+      GenesisRandomSeedBlockNumber,
+      GenesisRandomSeedPubKey,
+      EmptyBytesBlakeHash
+    )
```

Comment:
Please extract this part in a separate constructor function. I don't think we need to have these three constants at all in that case.